os/board/rtl8730e/src/component/wifi/inic/inic_ipc_host_trx.c

```diff
@@ -323,6 +329,10 @@ void inic_ipc_host_rx_handler(int idx_wlan, struct sk_buff *skb)
 	DCache_CleanInvalidate(((u32)skb - sizeof(struct list_head)), sizeof(struct skb_info));
 #endif /* CONFIG_ENABLE_CACHE */
 
+	if (p_buf) {
+		pbuf_free(p_buf);
+	}
```

Comment:
@ZhenBei-Sin You removed the comment only. But freeing `p_buf` at function success case is wrong. `p_buf` should be kept and should be used in recv buffer. Isn't it?

Reply:
function operation failure case
1. fail to alloc p_buf, go to the the RSP label, there is no free memory 
2. fail to alloc precvbuf, go to the RSP label, p_buf should be freed not to get memory leakage
They are ok with this code.


function operation success case (no error at anywhere)
1. after in-queue, go in the RSP label, it should not free any memory (p_buf, precvbuf) to use them later
but this new code frees p_buf in this case. then, your some code will use freed memory, then it causes memory corruption.